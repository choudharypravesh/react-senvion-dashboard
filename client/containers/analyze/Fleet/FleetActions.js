import AppConstants from '../../../constants/AppConstants';
import {GetData} from '../../../Utils/utils';
import _ from "underscore";

export  const changeFleetTab = (tabId) => dispatch => {
    dispatch({
        type: AppConstants.CHANGE_FLEET_TAB,
        payload:{tabId},
    })
}

//country and ID reference
// ID            HUB
// 1              america
// 3              eu_central
// 4              eu_north
// 5              eu_south_east
// 6              eu_south_west
// 2              australia
// 7              onshore
// 8              offshore
// 9              all
export const getFleetData = (data,selectedTab) => dispatch => {
    let self = this;
    dispatch({
        type: AppConstants.LOADING,
        category: data.id,
        payload: {
            loaderStatus: true,
            chartId: data.id,
            noData: false
        }
    });
    GetData.getFleetData(data,selectedTab).then(data => {
        if(_.isEmpty(data.allChartData)){
            dispatch({
                type: AppConstants.NODATA,
                category: data.id,
                payload: {
                    noData: true,
                    loaderStatus: false,
                    allChartData: {
                        X: [],
                        Y1: [],
                        Y2: [],
                        Y3: [],
                        Y4: [],
                        Y5: [],
                        name: "",
                        name1: "",
                        name2: "",
                        name3: "",
                        name4: "",
                        name5: ""
                    }
                }
            })
        }else{
            dispatch({
                type: AppConstants.LOADED,
                category: data.id,
                payload: {
                    loaderStatus: false,
                    noData: false,
                    allChartData: data.allChartData
                }
            });
        }
    }).catch(function(err) {
        console.log(err);
        window.alert("Faliure "+err);
    });

}
export const setDates = (initial, final, selectedDate,data,dataHub,selectedTab) => dispatch =>{
    dispatch({
        type: AppConstants.SET_DATES,
        payload: {
            initialDate: initial,
            finalDate: final,
            selectedDate: selectedDate
        }
    });
    if(data && dataHub) {
        dispatch(getFleetData(data, selectedTab))
        dispatch(getFleetData(dataHub, selectedTab))
    }
}