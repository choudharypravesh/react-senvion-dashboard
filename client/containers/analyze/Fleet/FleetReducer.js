import { combineReducers } from 'redux'
import moment from 'moment';
import {Map} from 'immutable'
import AppConstants from '../../../constants/AppConstants'


const initialState = Map({
    selectedTab: 1,
    selectedChart: 8,
    allDataLoader: true,
    allNoData: false,
    hubDataLoader: true,
    hubNoData: false,
    allChartData: {X:[], Y1:[], Y2:[], Y3:[], Y4:[], Y5:[], name:"", name1: "", name2: "", name3: "", name4: "", name5:""},
    hubChartData: {X:[], Y1:[], Y2:[], Y3:[], Y4:[], Y5:[], name:"", name1: "", name2: "", name3: "", name4: "", name5:""},
    initialDate: moment().subtract(7, 'days').format('YYYY-MM-DD'),
    finalDate: moment().format('YYYY-MM-DD'),
    selectedDate: 1
})

const fleetData = (state = initialState ,action) => {
    switch (action.type){
        case AppConstants.LOADING:
            if(action.category === 9){
                return state.set('allDataLoader',action.payload.loaderStatus).set('allNoData',action.payload.allNoData)
            }else{
                return state.set('hubDataLoader',action.payload.loaderStatus).set('selectedChart',action.payload.chartId).set('hubNoData',action.payload.noData)
            }
        case AppConstants.LOADED:
            if(action.category === 9){
                return state.set('allDataLoader',action.payload.loaderStatus).set('allChartData',action.payload.allChartData).set('allNoData',action.payload.noData)
            }else{
                return state.set('hubDataLoader',action.payload.loaderStatus).set('hubChartData',action.payload.allChartData).set('hubNoData',action.payload.noData)
            }
        case AppConstants.NODATA:
            if(action.category === 9){
                return state.set('allNoData',action.payload.noData).set('allDataLoader',action.payload.loaderStatus)
            }else{
                return state.set('hubNoData',action.payload.noData).set('hubDataLoader',action.payload.loaderStatus)
            }
        case AppConstants.CHANGE_FLEET_TAB:
            return state.set('selectedTab',action.payload.tabId)
        case AppConstants.SET_DATES:
            return state.set('initialDate',action.payload.initialDate).set('finalDate',action.payload.finalDate).set('selectedDate',action.payload.selectedDate)
        case AppConstants.GET_FLEET_DATA:
            return state;
        default:
            return state;
    }
}

export default fleetData;