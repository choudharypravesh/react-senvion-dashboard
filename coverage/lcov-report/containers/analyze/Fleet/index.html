<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for containers/analyze/Fleet</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="..\..\..\prettify.css" />
    <link rel="stylesheet" href="..\..\..\base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(..\..\..\sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="..\..\..\index.html">All files</a> containers/analyze/Fleet
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">60.31% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>79/131</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">71.64% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>48/67</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">57.41% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>31/54</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">60.31% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>79/131</span>
      </div>
    </div>
  </div>
  <div class='status-line medium'></div>
<div class="pad1">
<table class="coverage-summary">
<thead>
<tr>
   <th data-col="file" data-fmt="html" data-html="true" class="file">File</th>
   <th data-col="pic" data-type="number" data-fmt="html" data-html="true" class="pic"></th>
   <th data-col="statements" data-type="number" data-fmt="pct" class="pct">Statements</th>
   <th data-col="statements_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="branches" data-type="number" data-fmt="pct" class="pct">Branches</th>
   <th data-col="branches_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="functions" data-type="number" data-fmt="pct" class="pct">Functions</th>
   <th data-col="functions_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="lines" data-type="number" data-fmt="pct" class="pct">Lines</th>
   <th data-col="lines_raw" data-type="number" data-fmt="html" class="abs"></th>
</tr>
</thead>
<tbody><tr>
	<td class="file low" data-value="AvailabilityGen.js"><a href="AvailabilityGen.js.html">AvailabilityGen.js</a></td>
	<td data-value="40.43" class="pic low"><div class="chart"><div class="cover-fill" style="width: 40%;"></div><div class="cover-empty" style="width:60%;"></div></div></td>
	<td data-value="40.43" class="pct low">40.43%</td>
	<td data-value="47" class="abs low">19/47</td>
	<td data-value="57.69" class="pct medium">57.69%</td>
	<td data-value="26" class="abs medium">15/26</td>
	<td data-value="36.84" class="pct low">36.84%</td>
	<td data-value="19" class="abs low">7/19</td>
	<td data-value="40.43" class="pct low">40.43%</td>
	<td data-value="47" class="abs low">19/47</td>
	</tr>

<tr>
	<td class="file high" data-value="AvailabilityTime.js"><a href="AvailabilityTime.js.html">AvailabilityTime.js</a></td>
	<td data-value="93.02" class="pic high"><div class="chart"><div class="cover-fill" style="width: 93%;"></div><div class="cover-empty" style="width:7%;"></div></div></td>
	<td data-value="93.02" class="pct high">93.02%</td>
	<td data-value="43" class="abs high">40/43</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="26" class="abs high">26/26</td>
	<td data-value="84.21" class="pct high">84.21%</td>
	<td data-value="19" class="abs high">16/19</td>
	<td data-value="93.02" class="pct high">93.02%</td>
	<td data-value="43" class="abs high">40/43</td>
	</tr>

<tr>
	<td class="file high" data-value="Fleet.js"><a href="Fleet.js.html">Fleet.js</a></td>
	<td data-value="85.71" class="pic high"><div class="chart"><div class="cover-fill" style="width: 85%;"></div><div class="cover-empty" style="width:15%;"></div></div></td>
	<td data-value="85.71" class="pct high">85.71%</td>
	<td data-value="14" class="abs high">12/14</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="2" class="abs high">2/2</td>
	<td data-value="75" class="pct medium">75%</td>
	<td data-value="8" class="abs medium">6/8</td>
	<td data-value="85.71" class="pct high">85.71%</td>
	<td data-value="14" class="abs high">12/14</td>
	</tr>

<tr>
	<td class="file low" data-value="FleetStore.js"><a href="FleetStore.js.html">FleetStore.js</a></td>
	<td data-value="47.06" class="pic low"><div class="chart"><div class="cover-fill" style="width: 47%;"></div><div class="cover-empty" style="width:53%;"></div></div></td>
	<td data-value="47.06" class="pct low">47.06%</td>
	<td data-value="17" class="abs low">8/17</td>
	<td data-value="38.46" class="pct low">38.46%</td>
	<td data-value="13" class="abs low">5/13</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="2" class="abs high">2/2</td>
	<td data-value="47.06" class="pct low">47.06%</td>
	<td data-value="17" class="abs low">8/17</td>
	</tr>

<tr>
	<td class="file low" data-value="PortfolioGraphs.js"><a href="PortfolioGraphs.js.html">PortfolioGraphs.js</a></td>
	<td data-value="0" class="pic low"><div class="chart"><div class="cover-fill" style="width: 0%;"></div><div class="cover-empty" style="width:100%;"></div></div></td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="10" class="abs low">0/10</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="6" class="abs low">0/6</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="10" class="abs low">0/10</td>
	</tr>

</tbody>
</table>
</div><div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Tue Jul 18 2017 16:27:35 GMT+0530 (India Standard Time)
</div>
</div>
<script src="..\..\..\prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="..\..\..\sorter.js"></script>
</body>
</html>
