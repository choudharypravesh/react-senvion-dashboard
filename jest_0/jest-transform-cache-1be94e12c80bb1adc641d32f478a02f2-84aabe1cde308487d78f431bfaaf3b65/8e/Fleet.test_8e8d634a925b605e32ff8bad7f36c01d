'use strict';var _react = require('react');var _react2 = _interopRequireDefault(_react);
var _reactDom = require('react-dom');var _reactDom2 = _interopRequireDefault(_reactDom);
var _enzyme = require('enzyme');
var _reactTestRenderer = require('react-test-renderer');var _reactTestRenderer2 = _interopRequireDefault(_reactTestRenderer);
var _Fleet = require('../../../../client/containers/analyze/Fleet/Fleet');var _Fleet2 = _interopRequireDefault(_Fleet);
var _FleetContainer = require('../../../../client/containers/analyze/Fleet/FleetContainer');var _FleetContainer2 = _interopRequireDefault(_FleetContainer);
var _AppActions = require('../../../../client/actions/AppActions');var _AppActions2 = _interopRequireDefault(_AppActions);
var _CustomDatePicker = require('../../../../client/components/CustomDatePicker/CustomDatePicker');var _CustomDatePicker2 = _interopRequireDefault(_CustomDatePicker);
var _AvailabilityTime = require('../../../../client/containers/analyze/Fleet/AvailabilityTime');var _AvailabilityTime2 = _interopRequireDefault(_AvailabilityTime);

var _axios = require('axios');var _axios2 = _interopRequireDefault(_axios);
var _sinon = require('sinon');var _sinon2 = _interopRequireDefault(_sinon);function _interopRequireDefault(obj) {return obj && obj.__esModule ? obj : { default: obj };}