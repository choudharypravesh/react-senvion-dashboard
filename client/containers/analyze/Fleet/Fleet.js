// App.js
import React from 'react';
/*import 'react-dates/lib/css/_datepicker.css';*/
import 'react-virtualized-select/styles.css';
import classnames from 'classnames';
import {Link} from 'react-router'
import AvailabilityTime from './AvailabilityTime';
import AvailabilityGen from './AvailabilityGen';
import { connect } from 'react-redux'
import {setDates,getFleetData,changeFleetTab} from './FleetActions'
import CustomDatePicker from '../../../components/CustomDatePicker/CustomDatePicker';



class Fleet extends React.Component{
    constructor(props){
        super(props)
        this.getData = this.getData.bind(this);
        this.handleDateChangeFinal = this.handleDateChangeFinal.bind(this);
        this.handleDateChangeInitial = this.handleDateChangeInitial.bind(this);
        this.reloadGraphs = this.reloadGraphs.bind(this);
        this.updateDate=this.updateDate.bind(this);
    }

    getData(){
        switch(this.props.selectedTab){
            case 1:{
               let data= {
                   start_date: this.props.initialDate,
                   end_date: this.props.finalDate,
                   id: 9
               }
               let dataHub ={
                   start_date:this.props.initialDate,
                   end_date: this.props.finalDate,
                   id: this.props.selectedChart
               }
               return {data,dataHub}
            }
            case 2:{
                let data={
                    start_date: this.props.initialDate,
                    end_date: this.props.finalDate,
                    id: 9,
                    ranking: false,
                    level: 'hub',
                    variable: "pba"
                }
                let dataHub={
                    start_date:this.props.initialDate,
                    end_date: this.props.finalDate,
                    id: this.props.selectedChart,
                    ranking: false,
                    level: 'hub',
                    variable: "pba"
                }
                return {data,dataHub}
            }
        }
    }

    handleDateChangeInitial(initialDate) {
        let {dispatch} = this.props;
        dispatch(setDates(initialDate, this.props.finalDate, 0));
    }


    handleDateChangeFinal(finalDate) {
        const {dispatch} = this.props;
        dispatch(setDates(this.props.initialDate, finalDate, 0));
    }

    updateDate(start,end, selectedDate){
        const {dispatch} = this.props;
        let {data,dataHub} = this.getData();
        data.start_date=start;
        data.end_date=end;
        dataHub.start_date=start;
        dataHub.end_date=end;
        dispatch(setDates(start,end,selectedDate,data,dataHub,this.props.selectedTab));
    }

    reloadGraphs() {
        let {data,dataHub} = this.getData();
        const {dispatch} = this.props;
        dispatch(getFleetData(data,this.props.selectedTab))
        dispatch(getFleetData(dataHub,this.props.selectedTab))
    }

    getTabContent(){
        switch(this.props.selectedTab) {
            case 1 : return(
                <AvailabilityTime
                    initialDate={this.props.initialDate}
                    finalDate={this.props.finalDate}
                    selectedTab={this.props.selectedTab}
                    allChartData={this.props.allChartData}
                    allDataLoader={this.props.allDataLoader}
                    allNoData={this.props.allNoData}
                    hubChartData={this.props.hubChartData}
                    hubDataLoader={this.props.hubDataLoader}
                    hubNoData={this.props.hubNoData}
                    dispatch={this.props.dispatch}
                    selectedChart={this.props.selectedChart}
                />
            );
            case 2 : return(
                <AvailabilityGen
                    initialDate={this.props.initialDate}
                    finalDate={this.props.finalDate}
                    selectedTab={this.props.selectedTab}
                    allChartData={this.props.allChartData}
                    allDataLoader={this.props.allDataLoader}
                    allNoData={this.props.allNoData}
                    hubChartData={this.props.hubChartData}
                    hubDataLoader={this.props.hubDataLoader}
                    hubNoData={this.props.hubNoData}
                    dispatch={this.props.dispatch}
                    selectedChart={this.props.selectedChart}
                />
            );
        }
    }
    render() {
        return (
            <div className="portfolio-container" id="fleet-graphs">
                <div className="fleet-tabs margin-top level-1-tabs">
                    <div className="btn-group row" role="group">
                        <Link className={classnames("btn btn-default btn1 col-xs-4", {"active": this.props.selectedTab === 1})} onClick={() => {this.props.dispatch(changeFleetTab(1))}}>Availability(Time)</Link>
                        <Link className={classnames("btn btn-default btn2 col-xs-4", {"active": this.props.selectedTab === 2})} onClick={() => {this.props.dispatch(changeFleetTab(2))}}>Availability(Gen.)</Link>
                        <Link className="btn btn-default col-xs-4">Performance KPIs</Link>
                    </div>
                </div>
                <div className="row top-buffer">
                    <div className="col-xs-12">
                        <div className="font-12 flex-end-h flex-container-nowrap">
                            <CustomDatePicker ref = {(datePick) => { this.customDatePicker = datePick}}
                                              dateChange={this.updateDate}
                                              startOnChange={this.handleDateChangeInitial}
                                              endOnChange={this.handleDateChangeFinal}
                                              initialDate={this.props.initialDate}
                                              finalDate={this.props.finalDate}
                                              reloadGraphs={this.reloadGraphs}
                                              disabled={6}
                                              defaultValue={this.props.selectedDate}
                            />
                        </div>
                    </div>
                </div>
                {this.getTabContent()}
            </div>
        );
    }
}


const mapStateToProps = state => {
    const {FleetData} = state
    let Fleet = {
        selectedDate : FleetData.get('selectedDate'),
        finalDate : FleetData.get('finalDate'),
        initialDate : FleetData.get('initialDate'),
        selectedTab : FleetData.get('selectedTab'),
        selectedChart : FleetData.get('selectedChart'),
        allChartData : FleetData.get('allChartData'),
        allDataLoader : FleetData.get('allDataLoader'),
        allNoData : FleetData.get('allNoData'),
        hubChartData : FleetData.get('hubChartData'),
        hubDataLoader : FleetData.get('hubDataLoader'),
        hubNoData : FleetData.get('hubNoData'),
    }
    return Fleet
}
export default connect(mapStateToProps)(Fleet);
