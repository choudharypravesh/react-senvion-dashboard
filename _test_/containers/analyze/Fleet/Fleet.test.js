import React from 'react';
import ReactDOM from 'react-dom';
import { shallow, mount } from 'enzyme';
import renderer from 'react-test-renderer';
import Fleet from '../../../../client/containers/analyze/Fleet/Fleet';
import axios from 'axios'
import MockAdapter from 'axios-mock-adapter';
import moment from 'moment'

import { GetData } from '../../../../client/Utils/utils'



describe('Fleet', () => {

    var mock;
    var instance;
    let data = {start_date:"2017-06-10",end_date:"2017-07-10",ranking:false,level:"hub",variable:"pba",id:3}
    beforeEach(function() {
        instance = axios.create();
        mock = new MockAdapter(instance);
    });

    it('renders correctly', () => {
        const tree = renderer.create(
            <Fleet/>
        ).toJSON()
        expect(tree).toMatchSnapshot()
    })

    it('correctly sets the adapter on the axios instance', function() {
        expect(instance.defaults.adapter).toBeDefined();
    });

    it('Data is fetched and mapped correctly', function() {

        let data = {start_date:"2017-06-10",end_date:"2017-07-10",ranking:false,level:"hub",variable:"pba",id:3}
        mock.onGet('/api/get/fleet/availability?data='+JSON.stringify(data)).reply(200, [
            {date: "2017-06-10T00:00:00.000Z", type_broad: "MM", value: 88.3437593070246}]);

        return instance.get('/api/get/fleet/availability?data='+JSON.stringify(data)).then(function(response) {
            //console.log(response);
            expect(response.status).toBe(200);
            expect(response.data[0].type_broad).toBe('MM');
            expect(response.data[0].date).toBe('2017-06-10T00:00:00.000Z');
            expect(response.data[0].value).toBe(88.3437593070246);
        });
    });

    it('does not match when parameters are wrong', function() {
        mock
            .onGet('/api/get/fleet/availability', JSON.stringify(data))
            .reply(200);
        return instance
            .get('/api/get/fleet/availability', JSON.stringify(data))
            .catch(function(error) {
                expect(error.response.status).to.equal(404);
            });
    });

    it('Dates are in the selected range (Last 30 days)', function() {
        const FleetWrapper = mount(<Fleet/>);
        const button = FleetWrapper.find('#last30');
        button.simulate('click');

        let date1 = moment(Date.now()).format('YYYY-MM-DD');
        let date2 = moment(Date.now()).subtract(30, 'days').format('YYYY-MM-DD');
       // console.log(date1, date2)
        expect(FleetWrapper.state().data.get('initialDate')).toBe(date2);
        expect(FleetWrapper.state().data.get('finalDate')).toBe(date1);
    });

    it('Dates are in the selected range (Last 90 days)', function() {
        const FleetWrapper = mount(<Fleet/>);
        const button = FleetWrapper.find('#last90');
        button.simulate('click');

        let date1 = moment(Date.now()).format('YYYY-MM-DD');
        let date2 = moment(Date.now()).subtract(90, 'days').format('YYYY-MM-DD');
        //console.log(date1, date2)
        expect(FleetWrapper.state().data.get('initialDate')).toBe(date2);
        expect(FleetWrapper.state().data.get('finalDate')).toBe(date1);
    });

    it('Dates are in the selected range (Last 180 days)', function() {
        const FleetWrapper = mount(<Fleet/>);
        const button = FleetWrapper.find('#last180');
        button.simulate('click');

        let date1 = moment(Date.now()).format('YYYY-MM-DD');
        let date2 = moment(Date.now()).subtract(180, 'days').format('YYYY-MM-DD');
        //console.log(date1, date2)
        expect(FleetWrapper.state().data.get('initialDate')).toBe(date2);
        expect(FleetWrapper.state().data.get('finalDate')).toBe(date1);
    });

    it('Dates are in the selected range (Last 360 days)', function() {
        const FleetWrapper = mount(<Fleet/>);
        const button = FleetWrapper.find('#last360');
        button.simulate('click');
        let date1 = moment(Date.now()).format('YYYY-MM-DD');
        let date2 = moment(Date.now()).subtract(360, 'days').format('YYYY-MM-DD');
        //console.log(date1, date2)
        expect(FleetWrapper.state().data.get('initialDate')).toBe(date2);
        expect(FleetWrapper.state().data.get('finalDate')).toBe(date1);
    });

    it('Tab changes correctly on click', function() {
        const FleetWrapper = mount(<Fleet/>);
        const button = FleetWrapper.find('.btn2');
        expect(FleetWrapper.state().data.get('selectedTab')).toBe(1);
        button.simulate('click');
        expect(FleetWrapper.state().data.get('selectedTab')).toBe(2);
    });

    it('Go button click sends and recieves correct data', function() {
        const FleetWrapper = mount(<Fleet/>);
        let data = {start_date:"2017-06-10",end_date:"2017-07-10",ranking:false,level:"hub",variable:"pba",id:3}
        mock.onGet('/api/get/fleet/availability?data='+JSON.stringify(data)).reply(200, [
            {date: "2017-06-10T00:00:00.000Z", type_broad: "MM", value: 88.3437593070246}]);

        return instance.get('/api/get/fleet/availability?data='+JSON.stringify(data)).then(function(response) {
            console.log(response);
            const button = FleetWrapper.find('#submit-filter-selection');
            button.simulate('click');
            expect(response.status).toBe(200);
        });
    });

    it('Tab for differenet country should change correctly',() => {
        const CounrtyTab = mount (<Fleet/>);
        const buttons = CounrtyTab.find('#onshore');
        expect(CounrtyTab.state().data.get('selectedChart')).toBe(8);
        buttons.simulate('click');
        expect(CounrtyTab.state().data.get('selectedChart')).toBe(7);
        const buttons1 = CounrtyTab.find('#americas');
        expect(CounrtyTab.state().data.get('selectedChart')).toBe(7);
        buttons1.simulate('click');
        expect(CounrtyTab.state().data.get('selectedChart')).toBe(1);
        const buttons2 = CounrtyTab.find('#australia');
        expect(CounrtyTab.state().data.get('selectedChart')).toBe(1);
        buttons2.simulate('click');
        expect(CounrtyTab.state().data.get('selectedChart')).toBe(2);
        const buttons3 = CounrtyTab.find('#eu_south_east');
        expect(CounrtyTab.state().data.get('selectedChart')).toBe(2);
        buttons3.simulate('click');
        expect(CounrtyTab.state().data.get('selectedChart')).toBe(5);
        const buttons4 = CounrtyTab.find('#eu_central');
        expect(CounrtyTab.state().data.get('selectedChart')).toBe(5);
        buttons4.simulate('click');
        expect(CounrtyTab.state().data.get('selectedChart')).toBe(3);
        const buttons5 = CounrtyTab.find('#eu_north');
        expect(CounrtyTab.state().data.get('selectedChart')).toBe(3);
        buttons5.simulate('click');
        expect(CounrtyTab.state().data.get('selectedChart')).toBe(4);
        const buttons6 = CounrtyTab.find('#eu_south_west');
        expect(CounrtyTab.state().data.get('selectedChart')).toBe(4);
        buttons6.simulate('click');
        expect(CounrtyTab.state().data.get('selectedChart')).toBe(6);
    });
});



/*Need flux transformation for this testing*/
/*describe('The main app', () => {
    it('the app should have text', () => {
        const app  = shallow(<MainSkeleton/>);
        const userName = app.find('.user_name');
        expect(app.contains(<div>Hello jest from react</div>)).toBe(true);
    })
})*/



/*it('tab selection onshore ', function() {
    const FleetWrapper = mount(<Fleet/>);
    const button = FleetWrapper.find('.btn2');
    expect(FleetWrapper.state().data.get('selectedTab')).toBe(1);
    button.simulate('click');
    expect (FleetWrapper.state().data.get('selectedTab')).toBe(2);
});*/




// describe('<FleetContainer/>', () => {
//     it('renders 1 <FleetContainer/> component', () => {
//         const component = shallow(<FleetContainer name="fleet"/>);
//         expect(component).toHaveLength(1)
//     })
//     describe('it renders props correctly', () => {
//         const component = shallow(<FleetContainer name="fleet"/>)
//         console.log(component.instance().props);
//     })
//     describe('it changes the graphs on button click Go', () => {
//         const component = mount(<FleetContainer/>)
//         console.log(component);
//     })
// })



























