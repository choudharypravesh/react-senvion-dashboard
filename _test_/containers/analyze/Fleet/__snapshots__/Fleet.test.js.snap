// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Fleet renders correctly 1`] = `
<div
  className="portfolio-container"
  id="fleet-graphs"
>
  <div
    className="fleet-tabs margin-top level-1-tabs"
  >
    <div
      className="btn-group row"
      role="group"
    >
      <button
        className="btn btn-default btn1 col-xs-4 active"
        onClick={[Function]}
      >
        Availability(Time)
      </button>
      <button
        className="btn btn-default btn2 col-xs-4"
        onClick={[Function]}
      >
        Availability(Gen.)
      </button>
      <button
        className="btn btn-default col-xs-4"
      >
        Performance KPIs
      </button>
    </div>
  </div>
  <div
    className="fleet-graphs"
  >
    <div
      className="row top-buffer"
    >
      <div
        className="col-xs-12"
      >
        <div
          className="font-12 flex-end-h flex-container-nowrap"
        >
          <div
            className="flex-container-nowrap"
          >
            <div>
              <a
                className="btn btn-xs h-text disabled"
                id="today"
                onClick={[Function]}
              >
                Today
              </a>
               
              <a
                className="btn btn-xs h-text disabled"
                id="yesterday"
                onClick={[Function]}
              >
                Yesterday
              </a>
               
              <a
                className="btn btn-xs h-text selected"
                id="last7"
                onClick={[Function]}
              >
                Last 7 Days
              </a>
               
              <a
                className="btn btn-xs h-text"
                id="last30"
                onClick={[Function]}
              >
                Last 30 Days
              </a>
               
              <a
                className="btn btn-xs h-text"
                id="last90"
                onClick={[Function]}
              >
                Last 90 Days
              </a>
               
              <a
                className="btn btn-xs h-text"
                id="last180"
                onClick={[Function]}
              >
                Last 180 Days
              </a>
               
              <a
                className="btn btn-xs h-text"
                id="last360"
                onClick={[Function]}
              >
                Last 360 Days
              </a>
               
            </div>
            <div
              className="flex-container-nowrap flex-end-h date-picker"
            >
              <div
                className=""
              >
                <div
                  className="form-group"
                >
                  <div>
                    <div
                      style={
                        Object {
                          "display": "inline-block",
                        }
                      }
                    >
                      <div
                        className="react-datepicker__input-container"
                      >
                        <input
                          autoComplete={undefined}
                          autoFocus={undefined}
                          className=""
                          disabled={false}
                          id={undefined}
                          name={undefined}
                          onBlur={[Function]}
                          onChange={[Function]}
                          onClick={[Function]}
                          onFocus={[Function]}
                          onKeyDown={[Function]}
                          placeholder={undefined}
                          readOnly={undefined}
                          required={undefined}
                          tabIndex={undefined}
                          title={undefined}
                          type="text"
                          value="11/07/2017"
                        />
                      </div>
                    </div>
                  </div>
                </div>
              </div>
              <div>
                  To  
              </div>
              <div
                className=""
              >
                <div
                  className="form-group"
                >
                  <div>
                    <div
                      style={
                        Object {
                          "display": "inline-block",
                        }
                      }
                    >
                      <div
                        className="react-datepicker__input-container"
                      >
                        <input
                          autoComplete={undefined}
                          autoFocus={undefined}
                          className=""
                          disabled={false}
                          id={undefined}
                          name={undefined}
                          onBlur={[Function]}
                          onChange={[Function]}
                          onClick={[Function]}
                          onFocus={[Function]}
                          onKeyDown={[Function]}
                          placeholder={undefined}
                          readOnly={undefined}
                          required={undefined}
                          tabIndex={undefined}
                          title={undefined}
                          type="text"
                          value="18/07/2017"
                        />
                      </div>
                    </div>
                  </div>
                </div>
              </div>
              <div>
                    
              </div>
              <div
                className="date-go-button"
              >
                <button
                  className="btn btn-success btn-xs"
                  id="submit-filter-selection"
                  onClick={[Function]}
                >
                  Go
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="top-buffer"
    >
      <div
        className="loading"
        style={
          Object {
            "paddingBottom": "140px",
            "paddingLeft": "427px",
            "paddingRight": "427px",
            "paddingTop": "140px",
          }
        }
      >
        <svg
          height="20"
          width="70"
        >
          <circle
            cx="10"
            cy="10"
            r="0"
          >
            <animate
              attributeName="r"
              dur="1000ms"
              from="0"
              repeatCount="indefinite"
              to="3"
              values="0;3;3;3;0"
            />
          </circle>
          <circle
            cx="25"
            cy="10"
            r="0"
          >
            <animate
              attributeName="r"
              begin="200ms"
              dur="1000ms"
              from="0"
              repeatCount="indefinite"
              to="3"
              values="0;3;3;3;0"
            />
          </circle>
          <circle
            cx="40"
            cy="10"
            r="0"
          >
            <animate
              attributeName="r"
              begin="400ms"
              dur="1000ms"
              from="0"
              repeatCount="indefinite"
              to="3"
              values="0;3;3;3;0"
            />
          </circle>
        </svg>
      </div>
    </div>
    <div
      className="btn-group top-buffer-2"
      role="group"
    >
      <button
        className="btn btn-default active"
        id="offshore"
        onClick={[Function]}
      >
        Offshore
      </button>
      <button
        className="btn btn-default"
        id="onshore"
        onClick={[Function]}
      >
        Onshore
      </button>
      <button
        className="btn btn-default"
        id="americas"
        onClick={[Function]}
      >
        Americas
      </button>
      <button
        className="btn btn-default"
        id="australia"
        onClick={[Function]}
      >
        Australia
      </button>
      <button
        className="btn btn-default"
        id="eu_south_east"
        onClick={[Function]}
      >
        EU South East
      </button>
      <button
        className="btn btn-default"
        id="eu_central"
        onClick={[Function]}
      >
        EU Central
      </button>
      <button
        className="btn btn-default"
        id="eu_north"
        onClick={[Function]}
      >
        EU North
      </button>
      <button
        className="btn btn-default"
        id="eu_south_west"
        onClick={[Function]}
      >
        EU South West
      </button>
    </div>
    <div
      className="top-buffer-2"
      style={
        Object {
          "width": 924,
        }
      }
    >
      <div
        className="loading"
        style={
          Object {
            "paddingBottom": "140px",
            "paddingLeft": "427px",
            "paddingRight": "427px",
            "paddingTop": "140px",
          }
        }
      >
        <svg
          height="20"
          width="70"
        >
          <circle
            cx="10"
            cy="10"
            r="0"
          >
            <animate
              attributeName="r"
              dur="1000ms"
              from="0"
              repeatCount="indefinite"
              to="3"
              values="0;3;3;3;0"
            />
          </circle>
          <circle
            cx="25"
            cy="10"
            r="0"
          >
            <animate
              attributeName="r"
              begin="200ms"
              dur="1000ms"
              from="0"
              repeatCount="indefinite"
              to="3"
              values="0;3;3;3;0"
            />
          </circle>
          <circle
            cx="40"
            cy="10"
            r="0"
          >
            <animate
              attributeName="r"
              begin="400ms"
              dur="1000ms"
              from="0"
              repeatCount="indefinite"
              to="3"
              values="0;3;3;3;0"
            />
          </circle>
        </svg>
      </div>
    </div>
  </div>
</div>
`;
